src/main.cpp

```diff
@@ -3460,8 +3470,10 @@ void static CheckBlockIndex()
     CBlockIndex* pindexFirstNotScriptsValid = NULL; // Oldest ancestor of pindex which does not have BLOCK_VALID_SCRIPTS (regardless of being valid or not).
     while (pindex != NULL) {
         nNodes++;
+        // Not calling HaveBlockData() directly, since that can change state.
```

Comment:
Happy to change it, any suggestions?  The function only changes state if in pruning mode, so I don't want to go overboard with the description, and nothing concise is jumping out at me.

I did notice the comment for the function doesn't really emphasize the potentially state-changing behavior, I'll start by rewording that for now to make it more obvious.